packages/devtools_app/lib/src/screens/debugger/debugger_screen.dart

```diff
@@ -430,10 +430,21 @@ class _DebuggerStatusState extends State<DebuggerStatus> with AutoDisposeMixin {
       return 'running';
     }
 
-    final event = widget.controller.lastEvent!;
-    final frame = event.topFrame;
-    final reason =
-        event.kind == EventKind.kPauseException ? ' on exception' : '';
+    final event = widget.controller.lastEvent;
+    final String reason;
+    final Frame? frame;
+
+    if (event == null) {
+      // If paused from debugger screen, the event may be null.
+      // TODO(polina-c): if it is paused on exception first and then from debugger screen,
+      // we will give reason 'exception', that will be misleading.
+      // May be we should not give reason at all.
```

Comment:
This TODO is a bit confusing to read. It might be worth making this an issue and giving a more detailed description in that issue.